set.mm

```diff
@@ -788663,4 +788670,74 @@ to Davis and Putnam (1960).  (Contributed by David A. Wheeler,
 $( (End of Kunhao Zheng's mathbox.) $)
 $( End $[ set-mbox-kz.mm $] $)
 
+
+$( Begin $[ set-mbox-et.mm $] $)
+$( Skip $[ set-main.mm $] $)
+$(
+#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+  Mathbox for Ender Ting
+#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+$)
+
+
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  Increasing sequences and subsequences
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  $c UpWord $.
+
+  $( Extend class notation to include the set of increasing sequences. $)
+  cupword $a class UpWord S $.
+
+  ${
+    $d k w $.
+
+    ${
+      $( Increasing sequence is a sequence, adjacent elements of which
+         increase.  (Contributed by Ender Ting, 19-Nov-2024.) $)
+      df-upword $a |- UpWord S = { w | ( w e. Word S /\ A. k e. ( 0 ..^ ( ( # `
+      w ) - 1 ) ) ( w ` k ) < ( w ` ( k + 1 ) ) ) } $.
```

Comment:
Your definition is perfectly fine as is, but here is a remark anyway:
Since you are using `<`, and because `<` is only defined for the extended reals `RR*`, it will only ever apply for `S` being a subset of the extended real numbers `RR*`. 

Maybe you could simply drop the `S` and use `RR*` in the definition instead?

Reply:
Also, I think this should be described as a _strictly_ increasing sequence.

Reply:
I probably can't, as I intend to use this definition for more CS-related things, subsequence definition in particular; [element-order-preserving] subsequence of `S` would be represented as `UpWord ( 0 ..^ ( # ` S ) )` then.

Eventually, I want to prove a certain algorithm for finding longest increasing subsequence works correctly.

Reply:
Just to be sure I understand well:

In `A e. UpWord S`, `S` is the set where the values of `A` live, so ```B e. UpWord ( 0 ..^ ( # ` S ) )``` would be a sequence with values in ```( 0 ..^ ( # ` S ) )```, so an increasing sequence of _indices_. 
Then `( A o. B )`, the composition of `B` and `A` first maps to indices of `A`, then takes the value of `A`, so it is an element order preserving subsequence of `A`.

Am I correct?

If so, yes, this is a clever way to use the same symbol both for the sequences and for element-order-preserving subsequences.

Reply:
Exactly!